src/combinatorics/set_family/compression/uv.lean

```diff
@@ -0,0 +1,265 @@
+by rw [compression, compress_motion_self, compress_remains_self, empty_union]
+
+/-- Any family is compressed along two identical elements. -/
+lemma is_compressed_self (u : α) (s : finset α) : is_compressed u u s := compression_self u s
+
+lemma compress_disjoint (u v : α) : disjoint (compress_motion u v s) (compress_remains u v s) :=
+disjoint_left.2 $ λ a ha₁ ha₂, (mem_compress_motion.1 ha₁).1 (mem_compress_remains.1 ha₂).1
+
+/-- Compressing a set is idempotent. -/
+@[simp] lemma compress_idem (u v a : α) : compress u v (compress u v a) = compress u v a :=
+begin
+  unfold compress,
+  split_ifs with h h',
+  { suffices : u = ⊥,
+    { rw [this, sup_bot_eq, sup_bot_eq, _root_.sdiff_idem] },
+    rw ←disjoint_self,
+    have : u \ v = u := (h.1.mono_right h.2).sdiff_eq_left,
+    nth_rewrite 1 ←this,
+    exact h'.1.mono_right (sdiff_le_sdiff_right le_sup_right) },
+  { refl },
+  { refl }
+end
+
+lemma compress_mem_compression (ha : a ∈ s) : compress u v a ∈ 𝓒 u v s :=
+begin
+  rw mem_compression,
+  by_cases compress u v a ∈ s,
+  { rw compress_idem,
+    exact or.inl ⟨h, h⟩ },
+  { exact or.inr ⟨h, a, ha, rfl⟩ }
+end
+
+/-- Compressing a family is idempotent. -/
+@[simp] lemma compression_idem (u v : α) (s : finset α) : 𝓒 u v (𝓒 u v s) = 𝓒 u v s :=
+begin
+  have : ∀ a ∈ 𝓒 u v s, compress u v a ∈ 𝓒 u v s,
+  { intros a ha,
+    rw mem_compression at ⊢ ha,
+    simp only [compress_idem, exists_prop],
+    obtain ⟨_, ha⟩ | ⟨_, b, hb, rfl⟩ := ha,
+    { exact or.inl ⟨ha, ha⟩ },
+    { exact or.inr ⟨by rwa compress_idem, b, hb, (compress_idem _ _ _).symm⟩ } },
+  have : filter (λ a, compress u v a ∉ 𝓒 u v s) (𝓒 u v s) = ∅,
+    rw ←filter_false (𝓒 u v s), apply filter_congr, simpa,
+  rw [compression, compress_motion, this, image_empty, union_comm,
+      compress_remains, ←this],
+  exact filter_union_filter_neg_eq _ (compression u v s),
+end
+
+/-- Compressing a set family doesn't change its size. -/
+lemma card_compression (u v : α) (s : finset α) : (𝓒 u v s).card = s.card :=
+begin
+  rw [compression, card_disjoint_union (compress_disjoint _ _), compress_motion,
+    card_image_of_inj_on, ←card_disjoint_union, union_comm, compress_remains,
+    filter_union_filter_neg_eq],
+  { rw [disjoint.comm, disjoint_iff_inter_eq_empty],
+    exact filter_inter_filter_neg_eq _ _ },
+  intros a ha b hb hab,
+  dsimp at hab,
+  rw [mem_coe, mem_filter] at ha hb,
+  rw compress at ha hab,
+  split_ifs at ha hab with has,
+  { rw compress at hb hab,
+    split_ifs at hb hab with hbs,
+    { have h : (a ⊔ u) \ v \ u ⊔ v = (b ⊔ u) \ v \ u ⊔ v,
+      { rw hab },
+      rwa [sdiff_sdiff_comm, has.1.symm.sup_sdiff_cancel_right,
+          sdiff_sdiff_comm, hbs.1.symm.sup_sdiff_cancel_right,
+          sdiff_sup_cancel has.2, sdiff_sup_cancel hbs.2] at h },
+    { exact (hb.2 hb.1).elim } },
+  { exact (ha.2 ha.1).elim }
+end
+
+/-- If `a` is in the compressed family and does move under compression, then the compressed version
```

Comment:
this docstring doesn't really make much sense to me

Reply:
Better now? It's a bit technical (and many things are called compress something) that I don't know how to describe it better.

Reply:
yes! sorry for my awful review comments, clearly needed to be more awake this morning...